### DISCUSSION TITLE
Detecting Layout Issues

### DISCUSSION THREAD

**Discussion by girlpunk** [QuestPDF/QuestPDF#741] [Q&A] [answered]
Given `DocumentLayoutException` no longer exists as of 2023.10, is there a way to detect layout issues so users can be warned about problems? I'm thinking about a rather large document I have (200+ pages) where users may not view every page before submitting the document.

**Comment by MarcinZiabek** [comment, marked as answer]
The `QuestPDF.Settings.EnableDebugging` determines whether the layout issue should be annotated visually (by default: when the debugger is attached) or an appropriate layout exception is thrown (usually in production environments). 

Maybe, if the visual debugging is applied, the first page should be additionally annotated? What do you think?

**Reply by girlpunk** [reply]
Oh, I thought that might be the case and tried explicitly setting `Settings.EnableDebugging = false` in my startup, but still getting a PDF instead of an exception. Is that the correct way to test this when running from Visual Studio?

**Reply by girlpunk** [reply]
Ignore me, we had that being set to true later in the program to allow an ElementTrace to be generated to point users in the right direction with some workarounds. Is there a way to get the ElementTrace still, or is that gone?

**Reply by MarcinZiabek** [reply]
ElementTrace has been discontinued due to certain imprecisions I have found during the implementation of the layout exception visualization improvements. It will be introduced back as part of the new QuestPDF Companion effort, both as interactive UI and as the text-based trace. I also hope to provide significantly more details of why certain layout operations are performed. https://github.com/QuestPDF/QuestPDF/discussions/621

**Comment by DawidRobertNowak** [comment]
Hello,

We are closing this thread due to inactivity. We apologize for the delayed response. Thank you for taking the time to reach out and contribute to the project — your feedback helps us improve QuestPDF.

Should you encounter any further problems, please don't hesitate to open a new issue.

Thank you!
